Cargo.toml

```diff
@@ -119,3 +119,37 @@ thiserror = { version = "1.0.57" }
 trybuild = { version = "1.0.89" }
 toml = { version = "0.7.8" }
 url = { version = "2.3.1" }
+
+[workspace.lints.rust]
+unused_crate_dependencies = "warn"
+unused_lifetimes = "warn"
+unused_macro_rules = "warn"
+meta_variable_misuse = "warn"
+# Rust 2018 idioms that are not yet warn-by-default:
+elided_lifetimes_in_paths = "warn"
+unused_extern_crates = "warn"
+explicit_outlives_requirements = "warn"
+
+[workspace.lints.clippy]
+# Lints that are enabled (warn/deny) by default
+all = "warn"
+# Cargo
+cargo = "warn"                                              # Warn about Cargo.toml issues, except...
+multiple_crate_versions = { level = "allow", priority = 1 } # Not possible to deduplicate, should be done periodically ourselves
+# Restriction (optional, neutral lints)
+exit = "warn"                            # Prefer not `process::exit`ing directly
+rest_pat_in_fully_bound_structs = "warn" # Prefer not to use `..` in fully bound structs
+verbose_file_reads = "warn"              # Prefer simpler and more concise `fs::read_to_string`
+# # Pedantic
```

Comment:
nit: a few double comments, here and below:

Reply:
good catch!